frontend/src/model/beta/accessRequests/AccessRequestFormEditPage.tsx

```diff
@@ -0,0 +1,127 @@
+import { Box, Button, Divider, Stack, Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { patchAccessRequest, useGetAccessRequest } from 'actions/accessRequest'
+import { useContext, useEffect, useState } from 'react'
+import UnsavedChangesContext from 'src/contexts/unsavedChangesContext'
+import { getErrorMessage } from 'utils/fetcher'
+
+import { useGetSchema } from '../../../../actions/schema'
+import { useGetUiConfig } from '../../../../actions/uiConfig'
+import { AccessRequestInterface, SplitSchemaNoRender } from '../../../../types/interfaces'
+import { getStepsData, getStepsFromSchema } from '../../../../utils/beta/formUtils'
+import Loading from '../../../common/Loading'
+import ModelCardForm from '../../../Form/beta/ModelCardForm'
+import MessageAlert from '../../../MessageAlert'
+
+type FormEditPageProps = {
+  accessRequest: AccessRequestInterface
+}
+
+export default function AccessRequestFormEditPage({ accessRequest }: FormEditPageProps) {
+  const [isEdit, setIsEdit] = useState(false)
+  const [splitSchema, setSplitSchema] = useState<SplitSchemaNoRender>({ reference: '', steps: [] })
+  const [errorText, setErrorText] = useState('')
+
+  const { schema, isSchemaLoading, isSchemaError } = useGetSchema(accessRequest.schemaId)
+  const { mutateAccessRequest } = useGetAccessRequest(accessRequest.modelId, accessRequest.id)
+  const { uiConfig: _uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
+
+  const { setUnsavedChanges } = useContext(UnsavedChangesContext)
+  const theme = useTheme()
+
+  async function onSubmit() {
+    if (schema) {
+      const data = getStepsData(splitSchema, true)
+      const res = await patchAccessRequest(accessRequest.modelId, accessRequest.id, data)
+      if (!res.ok) {
+        setErrorText(await getErrorMessage(res))
+      } else {
+        setIsEdit(false)
+        setErrorText('')
+        mutateAccessRequest()
+      }
+    }
+  }
+
+  function onCancel() {
+    if (schema) {
+      mutateAccessRequest()
+      const steps = getStepsFromSchema(schema, {}, ['properties.contacts'], accessRequest.metadata)
+      for (const step of steps) {
+        step.steps = steps
+      }
+      setSplitSchema({ reference: schema.id, steps })
+      setIsEdit(false)
+    }
+  }
+
+  useEffect(() => {
+    if (!accessRequest || !schema) return
+    const metadata = accessRequest.metadata
+    const steps = getStepsFromSchema(schema, {}, ['properties.contacts'], metadata)
+
+    for (const step of steps) {
+      step.steps = steps
+    }
+
+    setSplitSchema({ reference: schema.id, steps })
+  }, [schema, accessRequest])
+
+  useEffect(() => {
+    setUnsavedChanges(isEdit)
+  }, [isEdit, setUnsavedChanges])
+
+  if (isSchemaError) {
```

Comment:
We've already spoken about this, but I'm happy to update these to use `MultipleErrorWrapper` in my ticket, so feel free to leave as is for now :)